readthedocs/oauth/models.py

```diff
@@ -1,120 +1,144 @@
+                                   related_name='oauth_repositories')
+    organization = models.ForeignKey(
+        RemoteOrganization, verbose_name=_('Organization'),
+        related_name='repositories', null=True, blank=True)
     active = models.BooleanField(_('Active'), default=False)
-    json = models.TextField('JSON')
 
-    objects = BitbucketProjectManager()
+    name = models.CharField(_('Name'), max_length=255)
+    full_name = models.CharField(_('Full Name'), max_length=255)
+    description = models.TextField(_('Description'), blank=True, null=True,
+                                   help_text=_('Description of the project'))
+    avatar_url = models.URLField(_('Owner avatar image URL'), null=True,
+                                 blank=True)
+
+    ssh_url = models.URLField(_('SSH URL'), max_length=512, blank=True,
+                              validators=[URLValidator(schemes=['ssh'])])
+    clone_url = models.URLField(
+        _('Repository clone URL'),
+        max_length=512,
+        blank=True,
+        validators=[
+            URLValidator(schemes=['http', 'https', 'ssh', 'git', 'svn'])])
+    html_url = models.URLField(_('HTML URL'), null=True, blank=True)
+
+    private = models.BooleanField(_('Private repository'), default=False)
+    admin = models.BooleanField(_('Has admin privilege'), default=False)
+    vcs = models.CharField(_('vcs'), max_length=200, blank=True,
+                           choices=REPO_CHOICES)
+
+    source = models.CharField(_('Repository source'), max_length=16,
+                              choices=OAUTH_SOURCE)
+    json = models.TextField(_('Serialized API response'))
+
+    objects = RemoteRepositoryManager()
+
+    class Meta:
+        ordering = ['organization__name', 'name']
+        verbose_name_plural = 'remote repositories'
 
     def __unicode__(self):
-        return "Bitbucket Project: %s" % (self.html_url)
-
-    def is_private(self):
-        full_json = eval(self.json)
-        return full_json['is_private']
+        return "Remote repository: %s" % (self.html_url)
+
+    def get_serialized(self, key=None, default=None):
+        try:
+            data = json.loads(self.json)
+            if key is not None:
+                return data.get(key, default)
+            return data
+        except ValueError:
+            pass
+
+    def matches(self, user):
+        """Projects that exist with repository URL already"""
+
+        # TODO
+        # ghetto_repo = self.clone_url.replace('git://', '').replace('.git', '')
+        projects = (Project
+                    .objects
+                    .public(user)
+                    .filter(repo=self.clone_url))
+        #            .filter(Q(repo__endswith=ghetto_repo) |
+        #                    Q(repo__endswith=ghetto_repo + '.git')))
+        # if projects:
+        #    repo.matches = [project.slug for project in projects]
+        # else:
+        #    repo.matches = []
```

Comment:
Yea, I think we need explicit mapping, and doing real tracking/resolution would require forcing GH auth and requiring that, which I'm not super excited to do.